### PR TITLE
Provide Initial Types for React Native FBSDK

### PR DESCRIPTION
**Pull request by ifiokjr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/react-native-fbsdk/index.d.ts
Comment (RyanCavanaugh) [inline types/react-native-fbsdk/index.d.ts]: This doesn't appear to be a UMD module

File: types/react-native-fbsdk/index.d.ts
Comment (RyanCavanaugh) [inline types/react-native-fbsdk/index.d.ts]: FWIW this union type does nothing -- it's just going to be the same as `string`

Comment (RyanCavanaugh) [inline types/react-native-fbsdk/index.d.ts]: (Feel free to leave around for documentation purposes, just an observation)

File: types/react-native-fbsdk/index.d.ts
Comment (RyanCavanaugh) [inline types/react-native-fbsdk/index.d.ts]: Is this a typo?

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Just a few minor comments

File: types/react-native-fbsdk/index.d.ts
Comment (ifiokjr) [inline types/react-native-fbsdk/index.d.ts]: Good spot, I copied it directly from the [flow type definition](https://github.com/facebook/react-native-fbsdk/blob/f27d02831c35a14790b40cc5c1c2381d480b028f/js/FBAppEventsLogger.js#L37).

File: types/react-native-fbsdk/index.d.ts
Comment (ifiokjr) [inline types/react-native-fbsdk/index.d.ts]: I used the same shape as the [React](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts) library definition, which seemed to be the only way I could get it to work in my code. 

Is there a better way?

File: types/react-native-fbsdk/index.d.ts
Comment (ifiokjr) [inline types/react-native-fbsdk/index.d.ts]: Thanks, I think I'll leave it around. It's proved helpful (to me) for documentation and autocompletion.

File: types/react-native-fbsdk/index.d.ts
Comment (RyanCavanaugh) [inline types/react-native-fbsdk/index.d.ts]: You just shouldn't need line 12 at all. It only modifies what shows up in non-module files

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Comment by RyanCavanaugh** [comment]
Bot is broken...

**Comment by RyanCavanaugh** [comment]
Going to sign off on this just so I can understand how the GitHub review API works

**Review by RyanCavanaugh — COMMENTED** [review]
I'm commenting for some reason

**Review by RyanCavanaugh — APPROVED** [review]
This workflow is not intuitive

**Comment by RyanCavanaugh** [comment]
@ifiokjr sorry for the spam - please remove the requested line and I'll merge. Thanks!

**Comment by ifiokjr** [comment]
No worries. I've fixed the typo and removed the requested line.

File: types/react-native-fbsdk/index.d.ts
Comment (andy-ms) [inline types/react-native-fbsdk/index.d.ts]: Why ignore this rule? You can fix this by removing the `declare namespace FBSDK` wrapper and exporting everything from the top-level.

File: types/react-native-fbsdk/index.d.ts
Comment (andy-ms) [inline types/react-native-fbsdk/index.d.ts]: Comment doesn't match the type name

File: types/react-native-fbsdk/index.d.ts
Comment (andy-ms) [inline types/react-native-fbsdk/index.d.ts]: For a case like this it might make more sense to use an inner `namespace ShareDialog` with functions such as `function canShow(...)` nested inside of that.

File: types/react-native-fbsdk/index.d.ts
Comment (weswigham) [inline types/react-native-fbsdk/index.d.ts]: I agree, @ifiokjr, these types would be much more straightforward without the containing namespace. Since it's not merged with a function or class, it is not necessary.

File: types/react-native-fbsdk/index.d.ts
Comment (weswigham) [inline types/react-native-fbsdk/index.d.ts]: Yeah, this comment seems copy-pasted from the interface below. Could it be corrected?

File: types/react-native-fbsdk/index.d.ts
Comment (weswigham) [inline types/react-native-fbsdk/index.d.ts]: Especially if it can't be instantiated. However this is fine if the interface needs to be implemented or referenced somewhere as a type. Is this interface being used to organize (in which case, make it a namespace), or not?

**Review by weswigham — CHANGES_REQUESTED** [review]

**Comment by ifiokjr** [comment]
Thanks for the suggestions. I've made the requested changes and, hopefully, it's okay to be merged now.

**Review by weswigham — APPROVED** [review]

**Pull request merged**
